build-logic/project-plugins/src/main/kotlin/com.hedera.hashgraph.platform.publish.gradle.kts

```diff
@@ -0,0 +1,95 @@
+    // 1. It does not support configuration cache.
+    //    https://github.com/GoogleCloudPlatform/artifact-registry-maven-tools/issues/85
+    // 2. It does not interact well with the 'gradle-nexus.publish-plugin' plugin, causing:
+    //    'No staging repository with name sonatype created' during IDE import
+    publishing.repositories.remove(publishing.repositories.getByName("sonatype"))
+    apply(plugin = "com.google.cloud.artifactregistry.gradle-plugin")
+}
+
+publishing.publications.named<MavenPublication>("maven") {
+    pom.description =
+        "Swirlds is a software platform designed to build fully-distributed " +
+            "applications that harness the power of the cloud without servers. " +
+            "Now you can develop applications with fairness in decision making, " +
+            "speed, trust and reliability, at a fraction of the cost of " +
+            "traditional server-based platforms."
+
+    pom.developers {
+        developer {
+            name = "Platform Base Team"
+            email = "platform-base@swirldslabs.com"
+            organization = "Hedera Hashgraph"
+            organizationUrl = "https://www.hedera.com"
+        }
+        developer {
+            name = "Platform Hashgraph Team"
+            email = "platform-hashgraph@swirldslabs.com"
+            organization = "Hedera Hashgraph"
+            organizationUrl = "https://www.hedera.com"
+        }
+        developer {
+            name = "Platform Data Team"
+            email = "platform-data@swirldslabs.com"
+            organization = "Hedera Hashgraph"
+            organizationUrl = "https://www.hedera.com"
+        }
+        developer {
+            name = "Release Engineering Team"
+            email = "release-engineering@swirldslabs.com"
+            organization = "Hedera Hashgraph"
+            organizationUrl = "https://www.hedera.com"
+        }
+    }
+}
+
+publishing.repositories {
+    maven("artifactregistry://us-maven.pkg.dev/swirlds-registry/maven-prerelease-channel") {
+        name = "prereleaseChannel"
+    }
+    maven("artifactregistry://us-maven.pkg.dev/swirlds-registry/maven-develop-snapshots") {
+        name = "developSnapshot"
+    }
+    maven("artifactregistry://us-maven.pkg.dev/swirlds-registry/maven-develop-daily-snapshots") {
+        name = "developDailySnapshot"
+    }
+    maven("artifactregistry://us-maven.pkg.dev/swirlds-registry/maven-develop-commits") {
+        name = "developCommit"
+    }
+    maven("artifactregistry://us-maven.pkg.dev/swirlds-registry/maven-adhoc-commits") {
+        name = "adhocCommit"
+    }
+}
+
+// Register one 'release*' task for each publishing repository
+publishing.repositories.all {
+    val ucName = name.replaceFirstChar { it.titlecase() }
+    tasks.register("release$ucName") {
+        group = "release"
+        dependsOn(tasks.named("publishMavenPublicationTo${ucName}Repository"))
+    }
```

Comment:
I removed duplication from this file by using a loop here. I also renamed the file and reformatted. That's why Git things this is all new (it isn't).